src/Firebase/AppCheck.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase;
+
+use Kreait\Firebase\AppCheck\ApiClient;
+use Kreait\Firebase\AppCheck\AppCheckToken;
+use Kreait\Firebase\AppCheck\AppCheckTokenGenerator;
+use Kreait\Firebase\AppCheck\AppCheckTokenOptions;
+use Kreait\Firebase\AppCheck\AppCheckTokenVerifier;
+use Kreait\Firebase\AppCheck\VerifyAppCheckTokenResponse;
+
+use function is_array;
+
+/**
+ * @internal
+ */
+final class AppCheck implements Contract\AppCheck
+{
+    private ApiClient $client;
+    private AppCheckTokenGenerator $tokenGenerator;
+    private AppCheckTokenVerifier $tokenVerifier;
+
+    public function __construct(
+        ApiClient $client,
+        AppCheckTokenGenerator $tokenGenerator,
+        AppCheckTokenVerifier $tokenVerifier
+    ) {
+        $this->client = $client;
+        $this->tokenGenerator = $tokenGenerator;
+        $this->tokenVerifier = $tokenVerifier;
+    }
```

Comment:
Just a note, you can resolve it instantly :D - it will be so satisfying to replace this with once support for PHP <8.1 is dropped 🤤 

```php
final class AppCheck implements Contract\AppCheck
{
    public function (
        private readonly ApiClient $client,
        private AppCheckTokenGenerator $tokenGenerator,
        private AppCheckTokenVerifier $tokenVerifier,
) {
    // …
}
```